app/views/play/level/WebSurfaceView.coffee

```diff
@@ -82,6 +86,53 @@ module.exports = class WebSurfaceView extends CocoView
     else
       deku.element(type, {}, children)
 
+  extractCssSelectors: (dekuStyles, html) ->
+    # TODO: Do something better than this hack for detecting which lines are CSS, which are HTML
+    dekuStyles.children.forEach (styleNode) =>
+      rawCss = styleNode.children[0].nodeValue
+      @rawCssLines ?= []
+      @rawCssLines = @rawCssLines.concat(rawCss.split('\n'))
+    @rawJQueryLines = _.filter (html.split('\n').map (line) -> (line.match(/^.*\$\(\s*['"].*['"]\s*\).*$/g) or [])[0])
+    # TODO: Move this hack for extracting CSS selectors
+    cssSelectors = _.flatten dekuStyles.children.map (styleNode) ->
+      rawCss = styleNode.children[0].nodeValue
+      try
+        parsedCss = parseCss(rawCss) # TODO: Don't put this in the global namespace
+        parsedCss.stylesheet.rules.map (rule) ->
+          rule.selectors.join(', ').trim()
+      catch e
+        # TODO: Report this error, handle CSS errors in general
+        []
+    # TODO: just do this in WebSurfaceView.onHoverLine?
+    # Find all calls to $("...")
+    jQuerySelectors = (html.match(/\$\(\s*['"](.*)['"]\s*\)/g) or []).map (jQueryCall) ->
+      # Extract the argument (because capture groups don't work with /g)
+      jQueryCall.match(/\$\(\s*['"](.*)['"]\s*\)/)[1]
+    Backbone.Mediator.publish('web-dev:extracted-css-selectors', { cssSelectors: cssSelectors.concat(jQuerySelectors) })
+    null
+
+  onExtractedCssSelectors: ({ @cssSelectors }) ->
+
+  onStopHoveringLine: ->
+    @iframe.contentWindow.postMessage({ type: 'highlight-css-selector', selector: '' }, '*')
+
+  onHoverLine: ({ row, line }) ->
+    if _.contains(@rawCssLines, line)
+      # They're hovering over lines of CSS, not HTML
+      trimLine = (line.match(/\s(.*)\s*{/)?[1] or line).trim().split(/ +/).join(' ')
+      hoveredCssSelector = _.find @cssSelectors, (selector) ->
+        trimLine is selector
+    else if _.contains(@rawJQueryLines, line)
+      # It's a jQuery call
+      trimLine = (line.match(/\$\(\s*['"](.*)['"]\s*\)/)?[1] or '').trim()
+      hoveredCssSelector = _.find @cssSelectors, (selector) ->
+        trimLine is selector
+    else
+      # They're not hovering over a line with a selector, so don't highlight anything
+      hoveredCssSelector = ''
+    @iframe.contentWindow.postMessage({ type: 'highlight-css-selector', selector: hoveredCssSelector }, '*')
+    null
```

Comment:
Does this only get emitted when moving onto a new line? If not, this function should probably be throttled.